yt_dlp/downloader/dash.py

```diff
@@ -87,4 +117,35 @@ def _get_fragments(self, fmt, ctx, extra_query):
                 'fragment_count': fragment.get('fragment_count'),
                 'index': i,
                 'url': fragment_url,
+                'decrypt_info': ctx.get('decrypt_info', {'METHOD': 'NONE'}),
             }
+
+    def _get_clearkey_cenc(self, info_dict, laurl, key_ids):
+        dash_cenc = info_dict.get('dash_cenc', {})
+        payload = json.dumps({
+            'kids': [
+                base64.urlsafe_b64encode(bytes.fromhex(k)).decode().rstrip('=')
+                for k in key_ids
+            ],
+            'type': 'temporary',
+        }).encode()
+        try:
+            response = self.ydl.urlopen(Request(
+                laurl, data=payload, headers={'Content-Type': 'application/json'}))
+            data = json.loads(response.read())
+        except (RequestError, json.JSONDecodeError) as err:
+            self.report_error(f'Failed to retrieve key from Clear Key license server: {err}')
+            return
+        keys = data.get('keys', [])
+        if len(keys) > 1:
+            self.report_warning('Clear Key license server returned multiple keys but only single key CENC is supported')
```

Comment:
There is nothing preventing us from supporting reading multiple keys from a Clear Key server (which returns a dict mapping of `{key_id: key, ...}` ) but ffmpeg currently only supports a single key per input file.